docs/rfds/sessions-list-api.mdx

```diff
@@ -0,0 +1,306 @@
+- `preview` (string, optional) - Snippet of the first user message or conversation preview
+- `_meta` (object, optional) - Agent-specific metadata (e.g., message count, error status, tags)
+
+#### Empty Result Example
+
+When no sessions match the criteria:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "result": {
+    "sessions": [],
+    "total": 0
+  }
+}
+```
+
+## Shiny future
+
+Once this feature exists:
+
+1. **Clients can build session browsers** - Users can view a list of all their conversations, sorted by recency or relevance
+2. **Session switching becomes seamless** - Users can easily switch between ongoing conversations
+3. **Better resource management** - Clients can identify and clean up old or inactive sessions
+4. **Cross-device continuity** - Users could potentially access their sessions from different devices (if agent supports it)
+5. **Improved UX patterns**:
+   - "Recent conversations" sidebar
+   - Search through past sessions
+   - Archive/delete old sessions
+   - Resume interrupted work easily
+
+Agents that implement this feature gain:
+- Better visibility into active sessions
+- Opportunity to implement session lifecycle policies
+- Foundation for future features like session sharing or collaboration
+
+## Implementation details and plan
+
+### Phase 1: Core Protocol Changes
+
+1. **Update schema.json** to add:
+   - `session/list` method definition
+   - `ListSessionsRequest` and `ListSessionsResponse` types
+   - `SessionInfo` type
+   - `listSessions` capability flag
+
+2. **Update protocol documentation** in `/docs/protocol/session-setup.mdx`:
+   - Document the new endpoint
+   - Explain when to use it vs. maintaining client-side session tracking
+   - Provide examples of common use cases
+
+### Phase 2: Reference Implementation
+
+3. **Implement in Rust SDK** (`rust/agent.rs` and `rust/client.rs`):
+   - Add `list_sessions` method to agent trait
+   - Provide default implementation (empty list) for agents without persistence
+   - Add client method to call `session/list`
+
+4. **Add to TypeScript SDKs** (if applicable):
+   - Update TypeScript types
+   - Add client methods
+
+### Phase 3: Example Implementation
+
+5. **Create example agent** that demonstrates:
+   - In-memory session registry
+   - Automatic title generation from first prompt
+   - Session lifecycle management (cleanup after N days)
+   - Pagination and filtering
+
+### Compatibility Considerations
+
+- **Backward compatible**: Existing agents continue working without implementing this
+- **Capability-based**: Clients check for `listSessions` capability before using
+- **No breaking changes**: No modifications to existing endpoints
+
+### Security Considerations
+
+- **Session isolation**: Agents must ensure sessions are only listed for the authenticated client
+- **Privacy**: The `preview` field should be optional and respect any privacy settings
```

Comment:
To this point, we could also start without the `preview` field and add it later?